OpenRA.Mods.Common/Traits/TransformOnCapture.cs

```diff
@@ -36,12 +39,28 @@ public TransformOnCapture(ActorInitializer init, TransformOnCaptureInfo info)
 
 		public void OnCapture(Actor self, Actor captor, Player oldOwner, Player newOwner)
 		{
+			if (!IsValidCaptor(captor))
+				return;
+
 			var facing = self.TraitOrDefault<IFacing>();
 			var transform = new Transform(self, info.IntoActor) { ForceHealthPercentage = info.ForceHealthPercentage, Faction = faction };
 			if (facing != null) transform.Facing = facing.Facing;
 			transform.SkipMakeAnims = info.SkipMakeAnims;
 			self.CancelActivity();
 			self.QueueActivity(transform);
 		}
+
+		bool IsValidCaptor(Actor captor)
+		{
+			var capInfo = captor.Info.TraitInfoOrDefault<CapturesInfo>();
+			if (capInfo != null && capInfo.CaptureTypes.Contains(info.Type))
+				return true;
+
+			var extCapInfo = captor.Info.TraitInfoOrDefault<ExternalCapturesInfo>();
+			if (extCapInfo != null && extCapInfo.CaptureTypes.Contains(info.Type))
```

Comment:
This can simplify down to `return extCapInfo != null && extCapInfo.CaptureTypes.Contains(info.Type)`.

Reply:
That is possible but this was deliberately done so to match capInfo lines.